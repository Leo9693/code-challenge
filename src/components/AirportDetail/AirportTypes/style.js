import styled, { keyframes } from "styled-components";
import { fadeInRight } from "react-animations";
export const FalseIcon = styled.span`
    padding: 5px;
    border-radius: 5px;
    color: red;
`;

export const TrueIcon = styled.span`
    padding: 5px;
    border-radius: 5px;
    color: green;
`;

export const FlexBox = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    transition: all 0.3s
    cursor:pointer
    &:hover {
        transform: scale(1.5);
        padding: 0 10px;
        border-radius: 20px;
        background-color: rgba(243, 181, 11, 0.6);
        box-shadow: 0 0 2px rgba(0, 0, 0, 0.3);
        & > span {
            background-color: rgba(243, 181, 11);
        }
    }
`;

export const AnimatedP = styled.p`
    transition: all 1s
    &:hover {
        transform: scale(1.1);
        box-shadow: 0 0 2px rgba(0, 0, 0, 0.3);
    }
`;
export const AnimatedAirportTypes = styled.div`
    animation: ${keyframes`${fadeInRight}`} 0.9s;
`;
