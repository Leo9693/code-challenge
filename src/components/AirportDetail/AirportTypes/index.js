import React from "react";
import { TrueIcon, FalseIcon, FlexBox } from "./style";
import { AnimatedAirportTypes } from "./style";
function airportType(props) {
    const {
        internationalAirport,
        domesticAirport,
        regionalAirport,
        onlineIndicator,
        eticketableAirport
    } = props.types;

    return (
        <AnimatedAirportTypes>
            <h3>Airport types:</h3>
            <FlexBox>
                InternationalAirport
                {internationalAirport ? (
                    <TrueIcon>
                        <i className="fas fa-check"></i>
                    </TrueIcon>
                ) : (
                    <FalseIcon>
                        <i className="fas fa-times"></i>
                    </FalseIcon>
                )}
            </FlexBox>
            <FlexBox>
                DomesticAirport
                {domesticAirport ? (
                    <TrueIcon>
                        <i className="fas fa-check"></i>
                    </TrueIcon>
                ) : (
                    <FalseIcon>
                        <i className="fas fa-times"></i>
                    </FalseIcon>
                )}
            </FlexBox>
            <FlexBox>
                RegionalAirport
                {regionalAirport ? (
                    <TrueIcon>
                        <i className="fas fa-check"></i>
                    </TrueIcon>
                ) : (
                    <FalseIcon>
                        <i className="fas fa-times"></i>
                    </FalseIcon>
                )}
            </FlexBox>
            <FlexBox>
                OnlineIndicator
                {onlineIndicator ? (
                    <TrueIcon>
                        <i className="fas fa-check"></i>
                    </TrueIcon>
                ) : (
                    <FalseIcon>
                        <i className="fas fa-times"></i>
                    </FalseIcon>
                )}
            </FlexBox>
            <FlexBox>
                EticketableAirport
                {eticketableAirport ? (
                    <TrueIcon>
                        <i className="fas fa-check"></i>
                    </TrueIcon>
                ) : (
                    <FalseIcon>
                        <i className="fas fa-times"></i>
                    </FalseIcon>
                )}
            </FlexBox>
        </AnimatedAirportTypes>
    );
}

export default airportType;
