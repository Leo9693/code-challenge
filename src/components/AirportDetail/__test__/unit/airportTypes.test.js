import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AirportTypes from "../../AirportTypes";
Enzyme.configure({ adapter: new Adapter() });

test("test ariport Types component", () => {
    const wrapper = shallow(
        <AirportTypes
            types={{
                internationAirport: "",
                domesticAirport: "",
                regionalAirport: "",
                onlineIndicator: "",
                eticketableAirport: ""
            }}
        />
    );
    expect(wrapper.debug()).toMatchSnapshot();
});
