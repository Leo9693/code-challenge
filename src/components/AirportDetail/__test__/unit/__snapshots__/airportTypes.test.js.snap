// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ariport Types component 1`] = `
"<styled.div>
  <h3>
    Airport types:
  </h3>
  <styled.div>
    InternationalAirport
    <styled.span>
      <i className=\\"fas fa-times\\" />
    </styled.span>
  </styled.div>
  <styled.div>
    DomesticAirport
    <styled.span>
      <i className=\\"fas fa-times\\" />
    </styled.span>
  </styled.div>
  <styled.div>
    RegionalAirport
    <styled.span>
      <i className=\\"fas fa-times\\" />
    </styled.span>
  </styled.div>
  <styled.div>
    OnlineIndicator
    <styled.span>
      <i className=\\"fas fa-times\\" />
    </styled.span>
  </styled.div>
  <styled.div>
    EticketableAirport
    <styled.span>
      <i className=\\"fas fa-times\\" />
    </styled.span>
  </styled.div>
</styled.div>"
`;
